doppler.js

```diff
@@ -1,15 +1,82 @@
 import https from "https";
 import { VERSION } from "./meta.js";
 
+// Adapted from https://github.com/DopplerHQ/cli/blob/a47f6d765bf9cb2155563e61287e325e492e3dea/pkg/http/config.go#L27
+const MAX_ATTEMPTS = 5;
+const BASE_DELAY = 500;
+
+/**
+ * Custom error class for Doppler API errors that the retry wrapper can use to determine if an error should be retried
+ */
+class DopplerApiError extends Error {
+  /**
+   * @param {string} message
+   * @param {number} statusCode
+   * @param {string} contentType
+   */
+  constructor(message, statusCode, contentType) {
+    super(message);
+    this.statusCode = statusCode;
+    this.contentType = contentType;
+  }
+}
+
+
 /**
- * Fetch secrets from Doppler the API
+ * Adapted from https://github.com/DopplerHQ/cli/blob/b008b823ec58f6982e26392b6fd0c30345e100a0/pkg/http/http.go#L359-L364
+ * Determines if a network error should be retried
+ */
+function shouldRetry(statusCode, contentType ) {
+  // don't retry 5xx errors w/ a JSON body
+  return (statusCode === 429 || (statusCode >= 100 && statusCode < 199) || (statusCode >=500 && statusCode <= 599 && !contentType?.startsWith("application/json")));
+
+}
+
+/**
+ * Retry wrapper with exponential backoff
+ * @param {Function} fn - The function to retry
+ * @param {number} maxAttempts - Maximum number of attempts
+ * @param {number} baseDelay - Base delay in ms
+ * @returns {Promise}
+ */
+async function withRetry(fn, maxAttempts = MAX_ATTEMPTS, baseDelay = BASE_DELAY) {
+  let lastError;
+  
+  for (let attempt = 0; attempt < maxAttempts; attempt++) {
+    try {
+      return await fn();
+    } catch (error) {
+      lastError = error;
+      
+      // Don't retry on the last attempt
+      if (attempt === maxRetries) {
```

Comment:
Looks like this got missed when the variable was named from `maxRetries` to `maxAttempts`.

```suggestion
      if (attempt === maxAttempts) {
```

Reply:
nice catch.... you are typescript